--- SDL_emscriptenevents.c	2021-03-23 00:30:38.756427025 +0200
+++ SDL_emscriptenevents.c	2021-03-23 00:30:52.380494951 +0200
@@ -138,7 +138,7 @@
     /*  92 */   SDL_SCANCODE_UNKNOWN,
     /*  93 */   SDL_SCANCODE_APPLICATION,
     /*  94 */   SDL_SCANCODE_UNKNOWN,
-    /*  95 */   SDL_SCANCODE_UNKNOWN,
+    /*  95 */   SDL_SCANCODE_ESCAPE,
     /*  96 */   SDL_SCANCODE_KP_0,
     /*  97 */   SDL_SCANCODE_KP_1,
     /*  98 */   SDL_SCANCODE_KP_2,
@@ -206,14 +206,14 @@
     /* 160 */   SDL_SCANCODE_GRAVE,
     /* 161 */   SDL_SCANCODE_UNKNOWN,
     /* 162 */   SDL_SCANCODE_UNKNOWN,
-    /* 163 */   SDL_SCANCODE_BACKSLASH,
+    /* 163 */   SDL_SCANCODE_KP_HASH, /*KaiOS phone keypad*/
     /* 164 */   SDL_SCANCODE_UNKNOWN,
     /* 165 */   SDL_SCANCODE_UNKNOWN,
     /* 166 */   SDL_SCANCODE_UNKNOWN,
     /* 167 */   SDL_SCANCODE_UNKNOWN,
     /* 168 */   SDL_SCANCODE_UNKNOWN,
     /* 169 */   SDL_SCANCODE_UNKNOWN,
-    /* 170 */   SDL_SCANCODE_UNKNOWN,
+    /* 170 */   SDL_SCANCODE_KP_MULTIPLY, /*KaiOS phone keypad*/
     /* 171 */   SDL_SCANCODE_RIGHTBRACKET,
     /* 172 */   SDL_SCANCODE_UNKNOWN,
     /* 173 */   SDL_SCANCODE_MINUS, /*FX*/
@@ -485,6 +485,22 @@
     if (keyEvent->keyCode < SDL_arraysize(emscripten_scancode_table)) {
         scancode = emscripten_scancode_table[keyEvent->keyCode];
 
+        if (keyEvent->keyCode == 0) {
+            /* KaiOS Left Soft Key and Right Soft Key, they act as OK/Next/Menu and Cancel/Back/Clear */
+            if (SDL_strncmp(keyEvent->key, "SoftLeft", 9) == 0) {
+                scancode = SDL_SCANCODE_AC_FORWARD;
+            }
+            if (SDL_strncmp(keyEvent->key, "SoftRight", 10) == 0) {
+                scancode = SDL_SCANCODE_AC_BACK;
+            }
+            if (SDL_strncmp(keyEvent->key, "Call", 4) == 0) {
+                scancode = SDL_SCANCODE_KP_ENTER;
+            }
+            if (SDL_strncmp(keyEvent->key, "EndCall", 7) == 0) {
+                scancode = SDL_SCANCODE_ESCAPE;
+            }
+        }
+
         if (scancode != SDL_SCANCODE_UNKNOWN) {
 
             if (keyEvent->location == DOM_KEY_LOCATION_RIGHT) {
